### DISCUSSION TITLE
Training neural SDF with loss functions that include input gradient.

### DISCUSSION THREAD

**Discussion by pettza** [NVlabs/instant-ngp#447] [Q&A]
Hello,

Thank you for sharing your work. I am interested in incorporating other kinds of losses in SDF training that include the input derivatives, for example the regularization loss proposed in this [paper](https://arxiv.org/abs/2002.10099). Is this possible to achieve with tcnn? Should I open a feature request there?

Thanks in advance

**Comment by Tom94** [comment]
Hi there, while tcnn doesn't _directly_ support losses w.r.t. input derivatives, @ventusff recently added support for double-backward in the case of the hash encoding: https://github.com/NVlabs/tiny-cuda-nn/pull/69

You could use this as a building block to extend support to tcnn's neural networks, or you could use it via PyTorch bindings in combination with a PyTorch-MLP.

Cheers!
